modules/ballerina-http/src/main/java/org/ballerinalang/net/ws/BallerinaWsServerConnectorListener.java

```diff
@@ -56,6 +58,8 @@ public void onMessage(WebSocketInitMessage webSocketInitMessage) {
         WebSocketService wsService = WebSocketDispatcher.findService(webSocketInitMessage);
         Resource onHandshakeResource = wsService.getResourceByName(Constants.RESOURCE_NAME_ON_HANDSHAKE);
         if (onHandshakeResource != null) {
+            Semaphore semaphore = new Semaphore(0);
+            AtomicBoolean isResourceExeSuccessful = new AtomicBoolean(false);
```

Comment:
why do we need an atomic boolean?

Reply:
We simply need a single object to change in inner class. Assigning new value inside inner class is not allowed. 
It gives the error "local variables referenced from an inner class must be final or effectively final"
So should I use another way to deal with it?

Reply:
Cant, we use Boolean. ?

Reply:
Ok. got it.